hypervisor/include/lib/list.h

```diff
@@ -108,8 +108,8 @@ static inline void list_splice_init(struct list_head *list,
 	}
 }
 
-#define list_entry(ptr, type, member) \
-	((type *)((char *)(ptr)-(uint64_t)(&((type *)0)->member)))
+#define container_of(ptr, type, member) \
+	((type *)((char *)(ptr)-(char *)(&((type *)0)->member)))
```

Comment:
Use 0 or NULL

Reply:
I prefer to use 0 instead of NULL. Here we want to calculate the offset of the member in the type structure. So we hope there is a type variable could point to address zero. We get the address of the member in it, we get the offset.

Reply:
ok

Reply:
Do not change (uint64_t) to (char *). The original code subtract an integer from a pointer, which results in a pointer of the same type. Changing the cast causes the subtraction to give an integer type and results in more kinds of type casts.

I'm fine with the rest. Personally I would prefer to use the `offsetof` macro (which is formally defined in C99 and implemented as __builtin_offsetof in gcc) rather than the tricky `&((type*)0->member)`. It's ok to me to preserve the current form if there is license concerns. Thanks.

Reply:
Thanks Junjie. Will use offsetof